evalml/pipelines/component_graph.py

```diff
@@ -0,0 +1,313 @@
+import networkx as nx
+import pandas as pd
+from networkx.algorithms.dag import topological_sort
+from networkx.exception import NetworkXUnfeasible
+
+from evalml.pipelines.components import ComponentBase, Estimator, Transformer
+from evalml.pipelines.components.utils import handle_component_class
+from evalml.utils import import_or_raise
+
+
+class ComponentGraph:
+    def __init__(self, component_dict=None, random_state=0):
+        """ Initializes a component graph for a pipeline as a directed acyclic graph (DAG).
+
+        Example:
+            >>> component_dict = {'imputer': ['Imputer'], 'ohe': ['One Hot Encoder', 'imputer.x'], 'estimator_1': ['Random Forest Classifier', 'ohe.x'], 'estimator_2': ['Decision Tree Classifier', 'ohe.x'], 'final': ['Logistic Regression Classifier', 'estimator_1', 'estimator_2']}
+            >>> component_graph = ComponentGraph(component_dict)
+           """
+        self.component_dict = component_dict or {}
+        for component_name, component_info in self.component_dict.items():
+            if not isinstance(component_info, list):
+                raise ValueError('All component information should be passed in as a list')
+            component_class = handle_component_class(component_info[0])
+            self.component_dict[component_name][0] = component_class
```

Comment:
Yep, you're understanding it right! I'm not sure I understand why differentiating this is necessary though, since this behavior follows how we currently handle `pipeline.component_graph` lists.